New Navigation and Settings

Clashsoft opened the pull request:
## New Features

+ Added a new Sidebar with recent polls and other useful navigation links.
+ Simplified the top Navbar by moving some elements to the Sidebar.
+ Added a new dedicated Settings page.

Closes #44 

# Screenshots

New Navigation Bar:

![image](https://github.com/Morphclue/apollusia/assets/4145923/3e74350e-0d67-42be-94e9-9343f0a0c81b)

New Settings:

<img src="https://github.com/Morphclue/apollusia/assets/4145923/62fd0088-1a36-41f1-aca6-47c79e8dfdf3" alt="New Settings screenshot" width=300>

New Sidebar:

<img src="https://github.com/Morphclue/apollusia/assets/4145923/982167ce-6c35-40fb-8666-fd7cdd66d8d5" alt="New Sidebar screenshot" width=300>

Morphclue commented on apps/frontend/src/app/core/navbar/navbar.component.html:
```diff
@@ -1,44 +1,98 @@
+        </a>
+      </li>
+      <div class="small text-muted text-center">
+        Copyright © Apollusia, 2024
```
Wouldn't it be better to implement the current year here instead of a hardcoded value that needs to be adjusted every year? (looking at you fulib 👀)

Clashsoft replied:
Yes, I will see if there is a good solution that does not just calculate the year on the user's time. I'd like it to be at build time

Morphclue reviewed (changes requested):
Just a few things I've noticed:
![image](https://github.com/Morphclue/apollusia/assets/43741877/c23ad275-e5db-4bf4-bf59-8afbb04dcb06)
Clicking on the sidebar highlights the X button.

A bit of spacing between the user icon and the dropdown icon wouldn't hurt.

![image](https://github.com/Morphclue/apollusia/assets/43741877/72b0f023-a617-42fa-a4f4-493b9facb827)
Currently has no valid route

Clashsoft commented:
>     1. Clicking on the sidebar highlights the X button.
This is the default ng-bootstrap behavior and I'm honestly ok with it. Keyboard only users and screenreader users can immediately exit the sidebar this way.
>     2. A bit of spacing between the user icon and the dropdown icon wouldn't hurt.
Fixed
>     3. Currently has no valid route
Fixed, but the behavior is not too good: The current form (e.g. New Poll) will be lost (not much worse than before though, because it used to reload anyway). I will come up with a fix in another PR.
